lib/util/git/index.ts

```diff
@@ -42,6 +43,52 @@ export { setPrivateKey } from './private-key';
 // TODO: fix upstream types https://github.com/steveukx/git-js/issues/704
 const ResetMode = (simpleGit.default as any).ResetMode as typeof _ResetMode;
 
+// Retry parameters
+const retryCount = 5;
+const delaySeconds = 3;
+const delayFactor = 2;
+
+// A generic wrapper for simpleGit.* calls to make them more fault-tolerant
+export async function gitRetry(
+  gitFunc: (...args) => Promise<any>,
+  ...args: any[] | undefined
+): Promise<any> {
```

Comment:
this is still too unsafe, as we loose all typescript type safty. Try something like:

```ts
export async function gitRetry<T>(
  gitFunc: () => Promise<T>
): Promise<T> {
// ...
      return await gitFunc();
// ...
}

// call like this:
await gitRetry(() => git.raw([
    'rev-parse',
    '--abbrev-ref',
    'origin/HEAD',
  ]));
```